src/compiler/transformer.ts

```diff
@@ -315,16 +351,13 @@ namespace ts {
                 }
 
                 if (lexicalEnvironmentVariableDeclarations) {
-                    const statement = createVariableStatement(
-                        /*modifiers*/ undefined,
-                        createVariableDeclarationList(lexicalEnvironmentVariableDeclarations)
-                    );
+                    const newStatements = map(chunkVariableDeclarationsBySyntheticComments(lexicalEnvironmentVariableDeclarations), createSimpleVariableStatement);
 
                     if (!statements) {

```

Comment:
This is exactly what `addRange` is for.
You could also just iterate over the results of `groupBy` instead of using `map`.